File: java/socketd/src/main/java/org/noear/socketd/transport/core/impl/ByteBufferReader.java
package org.noear.socketd.transport.core.impl;

import org.noear.socketd.transport.core.BufferReader;

import java.nio.ByteBuffer;

/**
 * @author noear
 * @since 2.0
 */
public class ByteBufferReader implements BufferReader {
    private ByteBuffer buffer;

    public ByteBufferReader(ByteBuffer buffer) {
        this.buffer = buffer;
    }

    @Override
    public byte get() {
        return buffer.get();
    }

    @Override
    public void get(byte[] dst, int offset, int length) {
        buffer.get(dst, offset, length);
    }

    @Override
    public int getInt() {
        return buffer.getInt();
    }

    @Override
    public int remaining() {
        return buffer.remaining();
    }

    @Override
    public int position() {
        return buffer.position();
    }
}

File: javascript/ts/org/noear/socketd/transport/core/buffer/ByteBufferReader.ts
import {BufferReader} from "./BufferReader";

/**
 * 缓冲读 ByteBuffer 适配
 *
 * @author noear
 * @since 2.0
 * @param {java.nio.ByteBuffer} buffer
 * @class
 */
export class ByteBufferReader implements BufferReader {
    /*private*/ buffer: ArrayBuffer

    public constructor(buffer: ArrayBuffer) {
        this.buffer = buffer;
    }

    public get(): number {
        return this.buffer
    }



    /**
     * 获取一组 byte
     * @param {byte[]} dst
     * @param {number} offset
     * @param {number} length
     */
    public getBytes(dst: number[], offset: number, length: number) {
        this.buffer.get(dst, offset, length);
    }

    /**
     * 获取 int
     * @return {number}
     */
    public getInt(): number {
        return this.buffer.getInt();
    }

    /**
     * 剩余长度
     * @return {number}
     */
    public remaining(): number {
        return this.buffer.remaining();
    }

    /**
     * 当前位置
     * @return {number}
     */
    public position(): number {
        return this.buffer.position();
    }
}